src/pages/index.js

```diff
@@ -0,0 +1,382 @@
+  selectedCardId = null
+  closeModal(deleteModal)
+})
+
+function openModal(modal) {
+  modal.classList.add('modal_is-opened')
+  document.addEventListener('keydown', handleEscClose)
+}
+
+function closeModal(modal) {
+  modal.classList.remove('modal_is-opened')
+  document.removeEventListener('keydown', handleEscClose)
+}
+
+function closeModalOnOverlayClick(evt) {
+  if (evt.target.classList.contains('modal')) {
+    closeModal(evt.target)
+  }
+}
+
+function handleEscClose(evt) {
+  if (evt.key === 'Escape') {
+    const openedModal = document.querySelector('.modal_is-opened')
+    if (openedModal) {
+      closeModal(openedModal)
+    }
+  }
+}
+
+function handleProfileEditClick() {
+  editProfileNameInput.value = profileNameEl.textContent
+  editProfileDescriptionInput.value = profileDescriptionEl.textContent
+  resetValidation(
+    editProfileForm,
+    [editProfileNameInput, editProfileDescriptionInput],
+    settings,
+  )
+  openModal(editProfileModal)
+}
+
+editProfileBtn.addEventListener('click', handleProfileEditClick)
+
+function handlePostModalOpen() {
+  openModal(newPostModal)
+}
+
+newPostBtn.addEventListener('click', handlePostModalOpen)
+
+editProfileModal.addEventListener('mousedown', closeModalOnOverlayClick)
+newPostModal.addEventListener('mousedown', closeModalOnOverlayClick)
+previewModal.addEventListener('mousedown', closeModalOnOverlayClick)
+
+const closeButtons = document.querySelectorAll('.modal__close-btn')
+closeButtons.forEach((button) => {
+  const modal = button.closest('.modal')
+  button.addEventListener('click', () => closeModal(modal))
+})
+
+function getNewPostValues() {
+  return {
+    name: newPostCaptionInput.value,
+    link: newPostImageInput.value,
+  }
+}
+
+function resetNewPostForm() {
+  newPostForm.reset()
+  const inputList = Array.from(
+    newPostForm.querySelectorAll(settings.inputSelector),
+  )
+  const buttonElement = newPostForm.querySelector(settings.submitButtonSelector)
+  toggleButtonState(inputList, buttonElement, settings)
+}
+
+function addNewCard(cardData, prepend = true) {
+  const cardEl = getCardEl(cardData)
+  if (prepend) {
+    cardsList.prepend(cardEl)
+  } else {
+    cardsList.append(cardEl)
+  }
+}
+
+function handleNewPostSubmit(evt) {
+  evt.preventDefault()
+  const submitButton = evt.submitter
+  setButtonText(submitButton, true, 'Create', 'Saving...')
+
+  const inputValues = getNewPostValues()
+  api
+    .addNewCard(inputValues)
+    .then((cardData) => {
+      addNewCard(cardData)
+      resetNewPostForm()
+      closeModal(newPostModal)
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false, 'Create', 'Saving...')
+    })
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault()
+  const submitButton = evt.submitter
+  setButtonText(submitButton, true)
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      const avatarImg = document.querySelector('.profile__avatar')
```

Comment:
`document.querySelector('.profile__avatar')` 

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine